packages/api-reference/src/components/CodeBlock/CodeBlock.vue

```diff
@@ -0,0 +1,186 @@
+<script lang="ts" setup>
+import { highlightElement, plugins } from 'prismjs'
+import 'prismjs/components/prism-bash'
+import 'prismjs/components/prism-json'
+import 'prismjs/plugins/autoloader/prism-autoloader.js'
+import 'prismjs/plugins/line-numbers/prism-line-numbers.css'
+import 'prismjs/plugins/line-numbers/prism-line-numbers.js'
+import { computed, nextTick, onMounted, ref, watch } from 'vue'
+
+/**
+ * Uses prism.js for syntax highlighting
+ * New languages are loaded async
+ *
+ * For lazy loading we can use the raw methods to load it sync
+ * const html = highlight(str, languages.javascript, 'javascript')
+ * + line numbers plugin etc etc
+ *
+ * @link https://prismjs.com/
+ */
+
+const props = withDefaults(
+  defineProps<{
+    content: string | object
+    lang?: string
+    lineNumbers?: boolean
+  }>(),
+  {
+    lang: 'js',
+    lineNumbers: false,
+  },
+)
+
+const el = ref(null)
+const language = computed(() => {
+  return props.lang === 'node' ? 'js' : props.lang
+})
+
+plugins.autoloader.languages_path =
+  'https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/'
```

Comment:
i know we are loading these async but should we have this bundled with the repo? :)

Reply:
I always prefer cdn over bundle for better performance! But doesn't matter too much these days

Reply:
alright im down, we might need to bundle it for some of our integrations but this is cool : )